proposals/interface-types/working-notes/scenarios/towints.md

```diff
@@ -0,0 +1,174 @@
+# Two Argument Integer Function
+
+Calling a two argument integer function, should result in effectively zero code.
+
+## Export
+
+```
+(@interface func (export "twizzle")
+  (param $a1 s32)(param $a2 s32) (result s32)
+  local.get $a1
+  s32-to-i23
+  local.get $a2
+  s32-to-i32
+  call "twizzle_"
+  i32-to-s32
+)
+```
+
+## Import
+
+```
+(@interface func (import "" "twozzle")
+  (param $a1 s32)(param $a2 s32) (result s32)
+)
+(@interface implement (import "" "twozzle_")
+  (param $b1 i32)(param $b2 i32) (result i32)
+  local.get $b1
+  i32-to-s32
+  local.get $b2
+  i32-to-s32
+  call-import "twozzle"
+  s32-to-i32
+)
+
+```
+
+## Adapter Code
+
+The adapter code, that maps the import of `twozzle_` to its implementation as
+`twizzle_` is:
+
+```
+(@adapter implement (import "" "twozzle_")
+  (param $b1 i32)(param $b2 i32) (result i32)
+    local.get $b1
+    local.get $b2
+    call Mx:"twizzle_"
+)
+```
+
+>Note: we adopt the convention that the `Mx:` prefix refers to the exporting
+>module and `Mi:` refers to the importing module.
```

Comment:
(look, I used the new multi-line comment feature!) To be symmetric with the [Lifting, lowering and laziness section](https://github.com/WebAssembly/interface-types/blob/master/proposals/interface-types/Explainer.md#lifting-lowering-and-laziness), it'd be good to write `call $twizzle_` instead of `call Mx:"twizzle_"`, showcasing that core exports cease to be exports after rewriting when they're only used by the adapted function.

Reply:
Agreed. I suggest using $Mx:twizzle_ to make it clear that it is a local function in the Mx module.

Reply:
I just checked and indeed ':' is a legal identifier character, so iiuc, what you're saying is that we just use a nice naming convention that makes the original module clear.  sgtm!